src/classes/CON_Preferred_Email_TEST.cls

```diff
@@ -0,0 +1,1977 @@
+        ));
+
+        Test.startTest();
+            tdtmClass.unsetRecursion();
+        Test.stopTest();
+
+        Boolean preferredTDTMFlag = TDTM_ProcessControl.getRecursionFlag(
+            TDTM_ProcessControl.registeredTrigger.CON_Preferred_TDTM
+        );
+
+        System.assertEquals(false, preferredTDTMFlag, 'CON_Preferred_TDTM should be Inactive');
+    }
+    
+    /********************************************************
+    * @description Tests isRecursion method
+    *********************************************************/
+    @isTest
+    static void isRecursion() {
+
+        CON_Preferred_TDTM tdtmClass = new CON_Preferred_TDTM();
+        tdtmClass.setRecursion();
+
+        System.assertEquals(true, TDTM_ProcessControl.getRecursionFlag(
+            TDTM_ProcessControl.registeredTrigger.CON_Preferred_TDTM
+        ));
+
+        Test.startTest();
+            tdtmClass.isRecursion();
+        Test.stopTest();
+
+        Boolean preferredTDTMFlag = TDTM_ProcessControl.getRecursionFlag(
+            TDTM_ProcessControl.registeredTrigger.CON_Preferred_TDTM
+        );
+
+        System.assertEquals(true, preferredTDTMFlag, 'CON_Preferred_TDTM should be Active');
+    }
+
+    /********************************************************************
+    * @description Tests isNewPreferredPhoneFunctionalityEnabled method
+    ********************************************************************/
+    @isTest
+    static void isNewPreferredPhoneFunctionalityEnabled() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c
+                                                         (Enable_New_Preferred_Phone_Sync__c = true));
+        
+        CON_Preferred_TDTM tdtmClass = new CON_Preferred_TDTM();
+
+        Test.startTest();
+            Boolean preferredPhoneEnabled = tdtmClass.isNewPreferredPhoneFunctionalityEnabled();
+        Test.stopTest();
+
+        System.assertEquals(true, preferredPhoneEnabled, 'Preferred Phone should be enabled');
+    }
+
+    /**************************************************************************************************************************
+    ****************************************************** FUNCTIONAL TESTS ***************************************************
+    **************************************************************************************************************************/
+
+    /*****************************************************************************************************************************
+    * @description Tests inserting a Contact with Preferred Email and no Email fields will throw PreferredEmailMatchNotNull Error
+    * when Preferred Email Enforcement is Enabled
+    *****************************************************************************************************************************/
+	@isTest 
+    static void preferredEmailNotBlankOnInsertWithEnforcementEnabled() {
```

Comment:
@skosgi1 Please update all of the functional/integration tests to use multiple contacts.